### PR TITLE
WorkspaceReaderSpec tests

### PR DESCRIPTION
**Pull request by marctalbott**
Two tests for tickets [QA-486](https://broadworkbench.atlassian.net/browse/QA-486) and [QA-533](https://broadworkbench.atlassian.net/browse/QA-533)

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Check that the **Product Owner** has signed off on any user-facing changes
- [x] **Submitter**: Make sure Swagger is updated if API changes
  - [ ] **...and Orchestration's Swagger too!**
- [x] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Database checks:
  * If PR includes new or changed db queries, include the explain plans in the description
  * Make sure liquibase is updated if appropriate
  * If doing a migration, take a backup of the
  [dev](https://console.cloud.google.com/sql/instances/terraform-qfarbdq3lrexxck5htofjs5z6m/backups?project=broad-dsde-dev&organizationId=548622027621)
  and
  [alpha](https://console.cloud.google.com/sql/instances/terraform-r4caezzc35c4tb7pgdhwkmme4y/backups?project=broad-dsde-alpha&organizationId=548622027621)
  DBs in Google Cloud Console
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] Tell your tech lead (TL) that the PR exists if they want to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [x] **TL** sign off
- [x] **LR** sign off
- [x] **Assign to submitter** to finalize
- [x] **Submitter**: Verify all tests go green, including CI tests
- [x] **Submitter**: Squash commits and merge to develop
- [x] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via Slack and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

**Review by aweng98 — APPROVED** [review]

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspaceApiSpec.scala [hunk 1]
```diff
@@ -244,6 +244,95 @@ class WorkspaceApiSpec extends TestKit(ActorSystem("MySpec")) with FreeSpecLike
...
+                eventually {
+                  val copyFromMethodRepoException = intercept[RestException] {
+                    Rawls.methodConfigs.copyMethodConfigFromMethodRepo(methodRepoConfig)(studentAToken)
+                  }
+                  assertExceptionStatusCode(copyFromMethodRepoException, 403)
+                }
+              }(ownerAuthToken)
+            }(ownerAuthToken)
+          }(ownerAuthToken)
+        }
```
Comment (ansingh7115) [inline automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspaceApiSpec.scala:313]: this feels a bit like 2 different tests - one for copying the mc from a workspace and importing the mc from the method repo.

Comment (marctalbott) [reply automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspaceApiSpec.scala:313]: agreed. I left it as one test so it'd be as close to the original FC-UI test as possible, but I think it makes more sense as two

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspaceApiSpec.scala [hunk 2]
```diff
@@ -244,6 +244,112 @@ class WorkspaceApiSpec extends TestKit(ActorSystem("MySpec")) with FreeSpecLike
...
+            withMethod("reader-import-from-method-repo", MethodData.SimpleMethod) { methodName =>
+              val method = MethodData.SimpleMethod.copy(methodName = methodName)
+
+              val destMethodConfig = Map(
+                "name" -> s"destination-${method.methodName}",
+                "namespace" -> s"destination-${method.methodNamespace}",
+                "workspaceName" -> Map(
+                  "namespace" -> projectName,
+                  "name" -> destWorkspaceName)
+              )
```
Comment (ansingh7115) [inline automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspaceApiSpec.scala]: `destMethodConfig` is defined in both tests, better to pull it out and put it either right above these two tests or the top of the class and only define it once

**Review by ansingh7115 — APPROVED** [review]
approved with small nit!

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspaceApiSpec.scala (hunk 2 above)
Comment (marctalbott) [reply automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspaceApiSpec.scala]: `projectName` and `destWorkspaceName` are both only defined within fixtures in the test, so I think unfortunately I have to leave `destMethodConfig` defined in both tests. I could pull it out into a function if you think that'd be better

**Pull request merged**